arch/risc-v/src/mpfs/mpfs_plic.c

```diff
@@ -213,3 +213,122 @@ uintptr_t mpfs_plic_get_thresholdbase(void)
 {
   return get_thresholdbase(riscv_mhartid());
 }
+
+/****************************************************************************
+ * Name: mpfs_plic_disable_irq(int extirq)
+ *
+ * Description:
+ *   Disable interrupt on all harts
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void mpfs_plic_disable_irq(int extirq)
+{
+  uintptr_t iebase;
+  uintptr_t claim_address;
+  int i;
+
+  /* Disable the irq on all harts */
+
+  for (i = 0; i < CONFIG_SMP_NCPUS; i++)
+    {
+      /* Clear any already claimed IRQ (this must be done BEFORE
+       * disabling the interrupt source):
+       *
+       * To signal the completion of executing an interrupt handler, the
+       * processor core writes the received interrupt ID to the
+       * Claim/Complete register. The PLIC does not check whether the
+       * completion ID is the same as the last claim ID for that target.
+       * If the completion ID does not match an interrupt source that is
+       * currently enabled for the target, the completion is ignored.
+       */
+
+      claim_address = mpfs_plic_get_claimbase(riscv_cpuid_to_hartid(i));
+      putreg32(extirq, claim_address);
+
+      /* Clear enable bit for the irq for every hart */
+
+      iebase = mpfs_plic_get_iebase(riscv_cpuid_to_hartid(i));
+      modifyreg32(iebase + (4 * (extirq / 32)), 1 << (extirq % 32), 0);
+    }
+}
+
+/****************************************************************************
+ * Name: mpfs_plic_clear_and_enable_irq
+ *
+ * Description:
+ *   Enable interrupt; if it is pending, clear it first
+ *
+ * Assumptions:
+ *   - Irq can only be pending, all the irq sources are disabled
+ *   - This and mpfs_plic_disable_irq are never called concurrently -
+ *     if they were, there would be mutual exclusion needed (spinlock)
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void mpfs_plic_clear_and_enable_irq(int extirq)
+{
+  int hartid = riscv_mhartid();
```

Comment:
Please use up_cpu_index, to make it easier to upstream this. riscv_mhartid has been deprecated.